pkg/apis/application/v1alpha1/repository_types.go

```diff
@@ -207,6 +207,26 @@ func (repo *Repository) GetGitCreds(store git.CredsStore) git.Creds {
 	return git.NopCreds{}
 }
 
+// GetCredentialType returns the type of the credentials used for the repository
+func (repo *Repository) GetCredentialType() string {
```

Comment:
I'd recommend moving this as a private utility function under commitserver/commit. I'd rather avoid exposing the string-name convention externally in case we decide to change it.

Reply:
Moved the function as mentioned.